src/azul/chalice.py

```diff
@@ -467,6 +483,33 @@ def swagger_resource(self, file) -> Response:
                                 headers={'Content-Type': content_type},
                                 body=body)
 
+    def threshold(self, *, errors: int, throttles: int):
+        def wrapper(f):
+            f.errors_threshold = errors
+            f.throttles_threshold = throttles
+            return f
+
+        return wrapper
+
+    def metric_thresholds(self) -> list[MetricThreshold]:
+        default_threshold = 0
+        thresholds = []
+        lambda_name, _ = config.unqualified_resource_name(self.app_name)
+        for metric in MetricThreshold.metrics:
+            # The api_handler lambda functions (indexer & service) aren't included
+            # in the app_module's handler_map, so we account for those first.
+            thresholds.append(MetricThreshold(lambda_name=lambda_name,
+                                              metric=metric,
+                                              value=default_threshold))
+            for handler in self.handler_map.values():
+                if isinstance(handler, chalice.app.EventSourceHandler):
+                    threshold = getattr(handler, f'{metric.lower()}_threshold', default_threshold)
```

Comment:
Encapsulate the naming rules in the enum. This can be modeled as a property or by parameterizing the enum instances.